camel/data_collector/alpaca_collector.py

```diff
@@ -0,0 +1,110 @@
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+
+from typing import Dict, List, Optional, Union
+
+from pydantic import BaseModel
+from typing_extensions import Self
+
+from camel.agents.chat_agent import ChatAgent
+from camel.data_collector.base import BaseDataCollector
+from camel.messages.base import BaseMessage
+from camel.schemas import OpenAISchemaConverter
+from camel.types.enums import OpenAIBackendRole
+
+DEFAULT_CONVERTER_PROMPTS = """
+    Extract key entities and attributes from the conversations
+    and convert them into a structured JSON format.
+    For example:
+    Instructions: You are a helpful assistant
+    User: When is the release date of the video game Portal?
+    Assistant: The release date of the video game Portal is October 9.
+    Your output should be:
+    {
+        "instructions": "You are a helpful assistant",
+        "input": "When is the release date of the video game Portal?",
+        "output": "The release date of the video game Portal is October 9."
+    }
+"""
+
+
+class AlpacaData(BaseModel):
+    instructions: str
+    input: str
+    output: str
```

Comment:
could we use the BaseModel from `camel/messages/conversion/alpaca.py` to reduce the redundancy